### PR TITLE
chore: revert multirm refactors

### PR DESCRIPTION
**Pull request by carolinaecalderon**
This reverts commit c3012ff9236f6a51758c72042c099980d3fe686d, 6857ecf20b899b1ee7837eebef68bf2ff0998a86,  7c6bec9d5de7eaf6e924693230814e0e0e637231.

## Description
Since we're now requiring globally unique resource pool names, we can remove the refactors that added an extra Resource Manager field to structs.

## Test Plan
Pass existing tests.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
RM-73

### CODE REVIEW

**Review by kkunapuli — APPROVED** [review]
LGTM

File: schemas/expconf/v0/resources.json [hunk 1]
```diff
@@ -52,13 +52,6 @@
             "maximum": 99,
             "default": null
         },
-        "resource_manager": {
```
Comment (wes-turner) [inline schemas/expconf/v0/resources.json:55]: I'd have expected that this results in an invalid experiment config that had a section
```yaml
resources:
  resource_manager: 'agent'
```
but when I try running an experiment with a section like this it seems to launch just fine. Is this patch backwards compatible in a way I don't understand?

Comment (carolinaecalderon) [reply schemas/expconf/v0/resources.json:55]: huh I'll double-check -- I ran 'git revert' on the three commits I wanted to revert, so this is unexpected

**Review by stoksc — APPROVED** [review]
lgtm

File: schemas/expconf/v0/resources.json (hunk 1 above)
Comment (stoksc) [reply schemas/expconf/v0/resources.json:55]: this change wasn't actually publicly released. does this cause any issues..

are there places with serialize then deserialize an experiment config that are going to pick this up and then explode?

Comment (stoksc) [reply schemas/expconf/v0/resources.json:55]: @carolinaecalderon to test this you may want to run an experiment before this revert, pause it, shutdown your cluster, add this revert, recompile, restart the cluster and see if resuming the experiment works.

if that works i think this is safe even though it isn't actually backward compatible.

Comment (carolinaecalderon) [reply schemas/expconf/v0/resources.json:55]: @wes-turner it's possible that when you tested this I hadn't re-generated the schemas, because when I run a test following @stoksc 's proposed flow, I get this error:
`ERRO[2024-03-06T15:15:46-05:00] failed to restore experiment: 1               error="cannot restore experiment 1 with unparsable config"`
tldr; the change is not backwards compatible since the experiment can't resume. However, since this change was not announced to the customer and is being reverted one release later, would it be fine to remove it?

Comment (stoksc) [reply schemas/expconf/v0/resources.json:55]: Wait no, that is exactly the error I was concerned about. You launched an experiment that did not use the multirm features and upgraded and it broke.

Comment (wes-turner) [reply schemas/expconf/v0/resources.json:55]: > this change wasn't actually publicly released. does this cause any issues..

Nice, nice, nice. No issues, no deserialization that I know about.

Thanks. The public deprecation was exactly my concern.

**Review by wes-turner — COMMENTED** [review]
LGTM

**Review by wes-turner — APPROVED** [review]

**Review by wes-turner — COMMENTED** [review]
Just BTW, I don't think we need backwards compatibility since this wasn't publicly released.

File: schemas/expconf/v0/resources.json (hunk 1 above)
Comment (stoksc) [reply schemas/expconf/v0/resources.json:55]: Turns out that doing this _does_ break experiment restore, so @carolinaecalderon is going to put up a fix. Really good you asked this @wes-turner, thanks.

**Comment by carolinaecalderon** [comment]
> Just BTW, I don't think we need backwards compatibility since this wasn't publicly released.

I think we do, because the original PR was included in 0.29.0 release. To fix this, I added back in the expconf changes and added a new db migration to drop the column

File: master/static/migrations/20240306164812_remove-rm-name-expconf.tx.down.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+UPDATE experiments
+SET config = config #- '{resource_manager}'
```
Comment (stoksc) [inline master/static/migrations/20240306164812_remove-rm-name-expconf.tx.down.sql:2]: this down migration doesn't make sense to me, can you explain it? i don't think the down migrations matter, but i'm curious.
